plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/update/VersionUpdateDialog.java

```diff
@@ -253,104 +252,109 @@ protected void buttonPressed(int buttonId)
-
-                            // Arm shutdown listener now because later will be too late
-                            final IWorkbenchListener listener = new IWorkbenchListener() {
-                                {
-                                    workbench.addWorkbenchListener(this);
-                                }
-
-                                @Override
-                                public boolean preShutdown(IWorkbench workbench, boolean forced) {
-                                    return true;
-                                }
-
-                                @Override
-                                public void postShutdown(IWorkbench workbench) {
-                                    try {
-                                        installer.run(file, log);
-                                    } catch (Exception e) {
-                                        log.error("Failed to run the installer script", e);
-                                    }
-                                }
-                            };
-
-                            UIUtils.asyncExec(() -> {
-                                ActionUtils.runCommand(IWorkbenchCommandConstants.FILE_EXIT, workbenchWindow);
-
-                                if (!workbench.isClosing()) {
-                                    workbench.removeWorkbenchListener(listener);
-                                    ShellUtils.launchProgram(folder.toString());
-                                }
-                            });
-                        } else {
-                            ShellUtils.showInSystemExplorer(file.toAbsolutePath().toString());
+                            installer.run(file, log);
+                        } catch (Exception e) {
+                            log.error("Failed to run the installer script", e);
                         }
-
-                        return Status.OK_STATUS;
                     }
                 };
-                job.setUser(true);
-                job.schedule();
-            } else {
-                ShellUtils.launchProgram(getDownloadPageURL(newVersion));
-            }
-        } else if (buttonId == CHECK_EA_ID) {
-            if (!CommonUtils.isEmpty(earlyAccessURL)) {
-                ShellUtils.launchProgram(earlyAccessURL);
-            }
-        } else if (buttonId == IDialogConstants.PROCEED_ID) {
-            final IWorkbenchWindow window = UIUtils.getActiveWorkbenchWindow();
-            CheckForUpdateAction.activateStandardHandler(window);
-            try {
-                ActionUtils.runCommand(CheckForUpdateAction.P2_UPDATE_COMMAND, PlatformUI.getWorkbench().getActiveWorkbenchWindow());
-            } finally {
-                CheckForUpdateAction.deactivateStandardHandler(window);
+
+                UIUtils.asyncExec(() -> {
+                    DBWorkbench.getPlatform().getPreferenceStore().setValue(DBeaverPreferences.CONFIRM_EXIT_TEMPORARY, true);
```

Comment:
This preference will be persisted on the disk. It might be possible that the value will stay `true` even though you reset it afterward.